packages/frontend-api/src/Model/Resolver/AutocompleteFavorites/AutocompleteFavoritesQuery.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrontendApiBundle\Model\Resolver\AutocompleteFavorites;
+
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+use Shopsys\FrameworkBundle\Model\Autocomplete\AutocompleteFavoriteFacade;
+use Shopsys\FrameworkBundle\Model\Product\ProductElasticsearchProvider;
+use Shopsys\FrontendApiBundle\Model\Resolver\AbstractQuery;
+
+class AutocompleteFavoritesQuery extends AbstractQuery
+{
+    protected const int PRODUCT_LIMIT = 5;
+    protected const int CATEGORY_LIMIT = 3;
+    protected const int BRAND_LIMIT = 3;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Autocomplete\AutocompleteFavoriteFacade $autocompleteFavoriteFacade
+     * @param \Shopsys\FrameworkBundle\Model\Product\ProductElasticsearchProvider $productElasticsearchProvider
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     */
+    public function __construct(
+        protected readonly AutocompleteFavoriteFacade $autocompleteFavoriteFacade,
+        protected readonly ProductElasticsearchProvider $productElasticsearchProvider,
+        protected readonly Domain $domain,
+    ) {
+    }
+
+    /**
+     * @return array{products: array[], categories: \Shopsys\FrameworkBundle\Model\Category\Category[], brands: \Shopsys\FrameworkBundle\Model\Product\Brand\Brand[]}
+     */
+    public function autocompleteFavoritesQuery(): array
+    {
+        $domainId = $this->domain->getId();
+
+        return [
+            'products' => $this->getSellableFavoriteProducts($domainId),
+            'categories' => $this->autocompleteFavoriteFacade->getCategoriesForDomain($domainId, self::CATEGORY_LIMIT),
+            'brands' => $this->autocompleteFavoriteFacade->getBrandsForDomain($domainId, self::BRAND_LIMIT),
+        ];
+    }
+
+    /**
+     * @param int $domainId
+     * @return array[]
+     */
+    protected function getSellableFavoriteProducts(int $domainId): array
+    {
```

Comment:
here is a slight inconsistency. The method is named getSellable, but inside it calls getListable. 
Personally, I would rather use sellable everywhere. I think the main point is to show the customer something he can buy.

Reply:
I wanted to be consistent with the regular searching - the only difference between listable and sellable is variants (the main variant is not sellable, while variants are not listable).

TLDR. In autocomple/search, there are always just main variants, so I want to keep this consistent even in the favorites box. I renamed the method in 25200c1c33

However, I am open to discussion here :slightly_smiling_face: